### PR TITLE
Kevin/load test cleanup

### PR DESCRIPTION
**Pull request by kevsul**
## Overview

Clean up and refactor load test code, making the scenarios more modular and composable.

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- Add FundAccount scenario
- Add SpendEthUtxo scenario
- Refactor duplicate submit_transaction code
- Allow a delay before sending transactions 
- Allow `gas_price` be configurable

### CODE REVIEW

File: priv/perf/apps/load_test/lib/child_chain/transaction.ex [hunk 1]
```diff
@@ -30,9 +52,16 @@ defmodule LoadTest.ChildChain.Transaction do
...
 
-    tx
-    |> Transaction.sign(keys: keys)
-    |> try_submit_tx(retries)
+    {:ok, blknum, txindex} =
+      tx
+      |> Transaction.sign(keys: keys)
+      |> try_submit_tx(retries)
+
+    outputs
```
Comment (boolafish) [inline priv/perf/apps/load_test/lib/child_chain/transaction.ex:60]: n00bie comment, but would it be more consistent for public api to return `{:ok, data}` instead?

File: priv/perf/config/stress.exs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 use Mix.Config
 
 config :load_test,
+  fee_wei: 30_000_000_000_000,
```
Comment (pgebal) [inline priv/perf/config/stress.exs:4]: Is this value correct?
Please remove the commented line.

File: priv/perf/apps/load_test/lib/child_chain/transaction.ex [hunk 3]
```diff
@@ -17,10 +17,32 @@ defmodule LoadTest.ChildChain.Transaction do
...
+
+  def spend_eth_utxo(utxo, amount, fee, sender, receiver, retries \\ 0) do
+    change_amount = utxo.amount - amount - fee
+    receiver_output = %Utxo{owner: receiver.addr, currency: @eth, amount: amount}
+    do_spend(utxo, receiver_output, change_amount, sender, retries)
+  end
+
+  defp do_spend(_input, _output, change_amount, _signer, _retries) when change_amount < 0, do: :error_insufficient_funds
+
+  defp do_spend(input, output, 0, signer, retries) do
```
[the remark this answers is not in the document]
Comment (kevsul) [reply priv/perf/apps/load_test/lib/child_chain/transaction.ex:38]: Ah, I meant to change it to `signer` everywhere. `signer` is used in `submit_tx` and I think it makes more sense

File: priv/perf/apps/load_test/lib/child_chain/transaction.ex [hunk 4]
```diff
@@ -17,10 +17,32 @@ defmodule LoadTest.ChildChain.Transaction do
...
+  alias ChildChainAPI.Model
   alias ExPlasma.Encoding
   alias ExPlasma.Transaction
+  alias ExPlasma.Utxo
+  alias LoadTest.Connection.ChildChain, as: Connection
 
   @retry_interval 1_000
+  @eth <<0::160>>
+
+  def spend_eth_utxo(utxo, amount, fee, sender, receiver, retries \\ 0) do
```
[the remark this answers is not in the document]
Comment (kevsul) [reply priv/perf/apps/load_test/lib/child_chain/transaction.ex:30]: Yeah, the problem is that if it's not eth then it has to be a fee token.
I guess we can allow it and add that as a comment

**Comment by kevsul** [comment]
> 1. What problem has you bumped so we start to configure `gas_price`? (Just curious, not saying it really needs justification)

Mainnet and Rinkeby gas prices are higher than Ropsten.

> 2. Same, why for the `delay`? Seems like you must encountered some instability?

No, this is so that you can control how fast the txs are sent. Setting this to 0 means send txs as fast as possible, but for example with a long running test we want to send txs at a slower rate

> 3. Might consider to add the essential config values in each scenario to their documents.

Yes, good idea

**Pull request merged**